tests/storage-volumes-vm

```diff
@@ -191,6 +191,74 @@ do
 	lxc storage volume detach "${poolName}" vol3 v1
 	lxc storage volume detach "${poolName}" vol6 v1 || true  # optional ISO
 
+	# attach VM root volumes
+	if hasNeededAPIExtension instance_root_volume_attachment; then
+		lxc init --empty --vm v2 --storage "${poolName}"
+		lxc init --empty --vm v3 --storage "${poolName}"
+
+		# Requires either security.shared or security.protection.start
+		! lxc storage volume attach "${poolName}" virtual-machine/v2 v1 || false
+
+		lxc config set v2 security.protection.start=true
+
+		# security.protection.start on a VM allows exactly one other attachment
+		lxc storage volume attach "${poolName}" virtual-machine/v2 v1
+		! lxc storage volume attach "${poolName}" virtual-machine/v2 v3 || false
+
+		# Deleting the instance will fail while it's root volume is in use
+		! lxc delete v2 || false
```

Comment:
I am now wondering if you can remove `v2`'s root volume and permanently attach it to `v1`? I think that's something we want for scenario where we "rescue" files from an unbootable VM.

The test is fine but that question that popped :)

Reply:
With this feature, the only meaningful distinction remaining between `virtual-machine`/`container` volumes and `custom` volumes is their lifecycle; that is, tied to instances vs independent. In the absence of that constraint (and any concerns around back-compat :sweat_smile: ), I'd advocate to remove the idea of storage volume types all together and just have "storage volumes".

If keeping the root's contents around without the instance is a capability that we want, it might be worth looking at whether creating a custom volume as a copy of an instance/snapshot is possible.